### DISCUSSION TITLE
Remembering last download location

### DISCUSSION THREAD

**Discussion by BookwyrmEDS** [aclap-dev/video-downloadhelper#665] [Help and Q&A]
### Your question

In the old version, when I chose 'SaveAs', it would remember the last place I downloaded a file to. Now I can only download to the default directory every time, and have to go searching. Am I missing the option in settings somewhere or was it removed? If it was removed, please put it back. 

### Addon & CoApp version and details

In the old version, when I chose 'SaveAs', it would remember the last place I downloaded a file to. Now I can only download to the default directory every time, and have to go searching. Am I missing the option in settings somewhere or was it removed? If it was removed, please put it back.

**Comment by paulrouget** [comment]
It's fixed in version 9.0.1.22!

**Reply by scarlion1** [reply]
In v 9.0.2.3, I don't see how to make it work.  I can set "download directory" in more settings but even this not seem to be respected and everything is automatically put into Firefox default download directory, even though Firefox setting "Always ask you where to save files" is activated and works for other files downloaded.

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I'm just another user.

In old VDH terminology, we used to have Quick Download.  In new VDH terminology, this is now called simply Download.  In old VDH terminology, we used to have Download.  In new VDH terminology, this is now called Download As.

Download uses the Download directory in the VDH settings.  Download As pops up the file selection dialog.  The trick is to find Download As when the default set when you first install VDH is Download (the old Quick Download, which did not give you a file selection dialog & same as Download still does not).

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/75f79600-32d1-452f-a7b6-cc7e6c7bd7b1)
![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/5a3428fa-f092-4ae1-bc2e-545afebb7c1e)

You have to do one Download As to prime the operation.  After just one Download As, VDH remembers whatever directory you choose & will use it next time you download something.  And so on forever after.  The Download directory in Settings is ONLY for Download.  Download As uses its own internal memory to remember where you most recently did a Download As to.  Old VDH used to update the Download directory setting after each download.  New VDH no longer does this.  Like I say, Download & Download As use 2 SEPARATE mechanisms for remembering where to download to.

The Firefox setting for the target directory of downloads is mostly irrelevant to VDH.  If you're not using the CoApp to do individual downloads (or all of them), then the Firefox setting gets involved.  But if you download via the CoApp -- using the CoApp is the recommended technique -- the Firefox setting does not participate.  Things work as I describe above.

**Reply by scarlion1** [reply]
Excellent
